ironfish/src/mining/pool.ts

```diff
@@ -390,4 +406,26 @@ export class MiningPool {
       decimalPrecision
     )
   }
+
+  async notifyStatus(): Promise<void> {
+    const status = await this.getStatus()
+    this.logger.debug(`Mining pool status: ${JSON.stringify(status)}`)
+    this.webhooks.map((w) => w.poolStatus(status))
+  }
+
+  async getStatus(): Promise<MiningPoolStatus> {
+    return {
+      name: this.name,
+      hashRate: await this.estimateHashRate(),
+      miners: this.stratum.clients.size,
+      sharesPending: await this.shares.sharesPendingPayout(),
+    }
+  }
+}
```

Comment:
You should put these 2 awaits in a Promise.all() to speed this up.

```suggestion
  async getStatus(): Promise<MiningPoolStatus> {
    const [hashRate, sharesPending] = await Promise.all([
        await this.estimateHashRate(),
        await this.shares.sharesPendingPayout(),
    ])

    return {
      name: this.name,
      hashRate: hashRate,
      miners: this.stratum.clients.size,
      sharesPending: sharesPending,
    }
  }
}
```